### PR TITLE
Expose public api

### PR DESCRIPTION
**Pull request by Fatme**

### CODE REVIEW

**Comment by teobugslayer** [comment]
Please bump the version in package.json

File: lib/ios-sim.ts [hunk 1]
```diff
@@ -14,3 +14,39 @@ var fiber = Fiber(() => {
...
+
+global.publicApi = {};
+
+Object.defineProperty(global.publicApi, "getRunningSimulator", {
+	get: () => {
+		return (...args: any[]) => {
+			let future = new Future<any>();
+			let timer = setInterval(() => {
+				Fiber(() => {
+					let libraryPath = require("./iphone-simulator-xcode-7");
```
Comment (teobugslayer) [inline lib/ios-sim.ts]: This can be moved out of the loop

File: lib/ios-sim.ts [hunk 2]
```diff
@@ -14,3 +14,39 @@ var fiber = Fiber(() => {
...
+					let obj = new libraryPath.XCode7Simulator()
```
Comment (teobugslayer) [inline lib/ios-sim.ts]: Can this be moved out of the loop too? I am not sure about this line.

File: lib/ios-sim.ts [hunk 3]
```diff
@@ -14,3 +14,39 @@ var fiber = Fiber(() => {
 });
 
 fiber.run();
+
+global.publicApi = {};
+
+Object.defineProperty(global.publicApi, "getRunningSimulator", {
```
Comment (teobugslayer) [inline lib/ios-sim.ts]: these are properties and should be named as nouns, not verbs. Consider naming it `runningSimulator`

File: lib/ios-sim.ts [hunk 4]
```diff
@@ -14,3 +14,39 @@ var fiber = Fiber(() => {
...
+						future.return(result);
+					}
+				}).run();
+			}, 500);
+			return future.wait();
+		}
+	}
+});
+
+Object.defineProperty(global.publicApi, "getApplicationPath", {
```
Comment (teobugslayer) [inline lib/ios-sim.ts]: these are properties and should be named as nouns, not verbs. Consider naming it `applicationPath`

File: lib/ios-sim.ts (hunk 2 above)
Comment (rosen-vladimirov) [inline lib/ios-sim.ts]: :+1:

File: lib/ios-sim.ts (hunk 1 above)
Comment (rosen-vladimirov) [inline lib/ios-sim.ts]: :+1:

File: lib/ios-sim.ts [hunk 5]
```diff
@@ -14,3 +14,39 @@ var fiber = Fiber(() => {
...
+Object.defineProperty(global.publicApi, "getRunningSimulator", {
+	get: () => {
+		return (...args: any[]) => {
+			let future = new Future<any>();
+			let timer = setInterval(() => {
+				Fiber(() => {
+					let libraryPath = require("./iphone-simulator-xcode-7");
+					let obj = new libraryPath.XCode7Simulator()
+					let result = obj.getBootedDevice.apply(obj, args).wait();
+					if(result && !future.isResolved()) {
```
Comment (rosen-vladimirov) [inline lib/ios-sim.ts]: in case device is never booted, we'll go in endless loop

Comment (tailsu) [inline lib/ios-sim.ts]: It's probably a good idea to give it some timeout, e.g. 15 seconds.

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**